From: Philipp Wiesemann <philipp.wiesemann@arcor.de>
Date: Fri, 5 Jun 2015 21:27:08 +0000 (+0200)
Subject: Fix warnings about using uninitialized variables
X-Git-Url: http://git.lxde.org/gitweb/?p=lxde%2Flibsysstat.git;a=commitdiff_plain;h=f72bc6173f25710089d237b1f37ff88b7772455d

Fix warnings about using uninitialized variables

Six warnings were given when compiling. Initializing the local variables
fixed them. It also fixed possibly emitting updates calculated by using
undefined values of the variables if reading from /proc/meminfo failed.
---

diff --git a/memstat.cpp b/memstat.cpp
index da20117..074fbc8 100644
--- a/memstat.cpp
+++ b/memstat.cpp
@@ -46,12 +46,12 @@ MemStatPrivate::~MemStatPrivate()
 
 void MemStatPrivate::timeout()
 {
-    qulonglong memTotal;
-    qulonglong memFree;
-    qulonglong memBuffers;
-    qulonglong memCached;
-    qulonglong swapTotal;
-    qulonglong swapFree;
+    qulonglong memTotal = 0;
+    qulonglong memFree = 0;
+    qulonglong memBuffers = 0;
+    qulonglong memCached = 0;
+    qulonglong swapTotal = 0;
+    qulonglong swapFree = 0;
 
     foreach (QString row, readAllFile("/proc/meminfo").split(QChar('\n'), QString::SkipEmptyParts))
     {
